### ISSUE TITLE
disabling syntax highlight return error for omnicompletion

### ISSUE THREAD

**Issue by zippeurfou** [jalvesaq/Nvim-R#208]
So when I disable the syntax highlight with: `let g:R_hi_fun = 0`
I am getting a couples of error before the omni completion get triggered:
![screen shot 2017-06-12 at 5 55 59 pm](https://user-images.githubusercontent.com/4520383/27057100-ac97abe4-4f98-11e7-986b-c935334bba3f.png)

This was triggered by typing `t.t` + `ctrl x` + `ctrl o`
What is strange is that after these warnings it actually does trigger some completion:
![screen shot 2017-06-12 at 6 00 48 pm](https://user-images.githubusercontent.com/4520383/27057202-1ce49cb8-4f99-11e7-9178-92200db88dd2.png)
My nvim config:
```
let g:R_nvimpager = 'horizontal'
let g:R_assign = 0
let g:R_indent_commented = 0
let g:R_commented_lines = 1
let g:R_listmethods = 1
let g:R_specialplot = 1
let g:R_show_args = 1
let g:R_tmux_split = 1
let g:R_hi_fun = 0
```
My vim version:
```
$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jun  6 2017 17:53:37)
MacOS X (unix) version
Included patches: 1-627
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     -tcl
-balloon_eval    +folding         +mouse_xterm     +termguicolors
-browse          -footer          +multi_byte      +terminfo
++builtin_terms  +fork()          +multi_lang      +termresponse
+byte_offset     -gettext         -mzscheme        +textobjects
+channel         -hangul_input    +netbeans_intg   +timers
+cindent         +iconv           +num64           +title
-clientserver    +insert_expand   +packages        -toolbar
+clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        +perl            +vertsplit
+cmdline_hist    +keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       +python          +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con      -lua             +rightleft       +writebackup
+diff            +menu            +ruby            -X11
+digraphs        +mksession       +scrollbind      -xfontset
-dnd             +modify_fname    +signs           -xim
-ebcdic          +mouse           +smartindent     -xpm
+emacs_tags      -mouseshape      +startuptime     -xsmp
+eval            +mouse_dec       +statusline      -xterm_clipboard
+ex_extra        -mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax
+farsi           +mouse_netterm   +tag_binary
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o vim        -lncurses -liconv -framework Cocoa   -mmacosx-version-min=10.12 -fstack-protector-strong -L/usr/local/lib  -L/usr/local/Cellar/perl/5.24.1/lib/perl5/5.24.1/darwin-thread-multi-2level/CORE -lperl -lm -lutil -lc -F/usr/local/opt/python/Frameworks -framework Python   -lruby.2.4.1 -lobjc
```

**Comment by jalvesaq** [comment]
To fix this issue, I have to change the syntax/r.vim from R-Vim-runtime...

**Comment by jalvesaq** [comment]
@zippeurfou Could you please install [R-Vim-runtime](https://github.com/jalvesaq/R-Vim-runtime) and update Nvim-R? The bug seems to be fixed here (using Neovim).

**Comment by zippeurfou** [comment]
It works by following your instructions.
As a side note I am disabling the highlight because the colors do not make much sense in my opinion. I might be missing something but it seems to me that anything that can get completed have the same color which is very confusing.
Here is what I mean.
Before executing anything
![screen shot 2017-06-13 at 10 45 24 am](https://user-images.githubusercontent.com/4520383/27088441-a8f4c8c2-5025-11e7-927c-ceafbe0165c2.png)
After executing till line 10
![screen shot 2017-06-13 at 10 46 24 am](https://user-images.githubusercontent.com/4520383/27088500-d4b4c78c-5025-11e7-9386-5e7b3abf4c87.png)
As you can see there are multiple things that are confusing to me:
1. `rpart` became yellow on line 6 which is not consistent with the other require blocks.
2. `data` is by default highlighted in yellow but `data_test` is not
3. some arguments get highlighted like `by` but not all which looks random to me.
I don't believe that all of this is a bug but it makes just very hard to follow the highlight in my opinion.

**Comment by jalvesaq** [comment]
It's not a bug. I like to avoid naming objects with existing valid names, and I also like to see the color of names of functions that I'm typing turning into another color because, then, I know that I did not misspelled them. That is I use syntax highlight to help me writing the code, and not only to read it. I agree with you in not liking names of non function objects highlighted as functions, but I prefer this to the currently available alternative: highlight anything followed by a parenthesis:
```
let g:r_syntax_fun_pattern = 1
```

**Comment by jalvesaq** [comment]
I have just added an option to highlight only names followed by `(` that are loaded in R memory (or that are in `R_start_libs`). Please, update Nvim-R and try:
```
let R_hi_fun_paren = 1
```

**Comment by zippeurfou** [comment]
I just updated it, didn't change the configuration. There might be an issue with the last commit.
![screen shot 2017-06-13 at 12 04 51 pm](https://user-images.githubusercontent.com/4520383/27092205-99c44098-5030-11e7-87a9-e0d9dc5098f9.png)

**Comment by zippeurfou** [comment]
Trying out different combination seems to have odd behavior.
conf 1.
![screen shot 2017-06-13 at 12 23 23 pm](https://user-images.githubusercontent.com/4520383/27092981-267e081e-5033-11e7-89d1-c0f85211e93b.png)
before and after  executing the lines are the same:
![screen shot 2017-06-13 at 12 22 15 pm](https://user-images.githubusercontent.com/4520383/27092980-26796458-5033-11e7-952b-fe6d0c3605c0.png)

conf 2 (turning on `R_hi_fun`).
![screen shot 2017-06-13 at 12 28 13 pm](https://user-images.githubusercontent.com/4520383/27093191-cbc29cd6-5033-11e7-89ec-e0d2a5e1ea34.png)

before executing:
![screen shot 2017-06-13 at 12 25 02 pm](https://user-images.githubusercontent.com/4520383/27093066-5eee1dec-5033-11e7-9269-9a864fa6ea22.png)
The `group_by()` highlight seems wrong to me.
after executing the lines (perfect 👍 ):
![screen shot 2017-06-13 at 12 27 08 pm](https://user-images.githubusercontent.com/4520383/27093160-aa2504a6-5033-11e7-8752-ebe4da7829e5.png)

**Comment by jalvesaq** [comment]
In some systems, Vim/Neovim sources syntax files before filetype files, that is, before `R_hi_fun_paren` was created.
I can see errors like the wrong highlighting of `group_by` when editing RMarkdown with the default `syntax/markdown.vim` script. The error disappears if I install [vim-pandoc-syntax](https://github.com/vim-pandoc/vim-pandoc-syntax).
The `magrittr` package has a function called `ne'est pas`.

**Comment by jalvesaq** [comment]
I have fixed some issues. Please, let me know if you find more bugs.

**Comment by zippeurfou** [comment]
Looking at it now it seems perfect. I think you can close this issue.
As a side not: one thing that I just noticed that is not directly related and is just interesting to know (maybe as a future feature) is that the omnicompletion is slightly different than Rstudio.
Library/Require is not as smart as in Rstudio
<img width="265" alt="screen shot 2017-06-13 at 10 35 09 pm" src="https://user-images.githubusercontent.com/4520383/27113098-f157342a-5088-11e7-8f2b-5390bede1bac.png">
<img width="765" alt="screen shot 2017-06-13 at 10 34 53 pm" src="https://user-images.githubusercontent.com/4520383/27113097-f156abcc-5088-11e7-84a7-0568fb56f843.png">
And when the function use "special" character Rstudio escape it with ``` meanwhile nvim does not:
<img width="324" alt="screen shot 2017-06-13 at 10 37 00 pm" src="https://user-images.githubusercontent.com/4520383/27113128-186e6e8e-5089-11e7-9fdc-d277cdff75d3.png">
<img width="407" alt="screen shot 2017-06-13 at 10 36 12 pm" src="https://user-images.githubusercontent.com/4520383/27113129-186f3814-5089-11e7-8139-1bdb865cdfbd.png">
Thanks a lot for this awesome plugin. I am now so excited to be able to use the awesomeness of R combined with vim :).

**Comment by jalvesaq** [comment]
Nvim-R also completes the names of packages for both `library()` and `require()` but only if R is running. However, it's not omnicompletion; it's function argument completion. Please, try Ctrl+X Ctrl+A.

**Comment by jalvesaq** [comment]
It's easy to escape names with special characters, but what special characters should I look for in addition to ` `, `'` and `%`?

**Comment by zippeurfou** [comment]
Yes I agree it is a bit tricky since every operation are function that you can change (i.e. `for`,`{`,`+`,`[]`). You also have the complete list of infix operators `:, ::, :::, $, @, ^, *, /, +, -, >, >=, <, <=, ==, !=, !, &, &&, |, ||, ~, <-, <<-`.

**Comment by jalvesaq** [comment]
I was expecting a smaller list, and I am afraid that it would be too complicated to support all these characters. So, now I am going to support only the three that I mentioned.

**Comment by zippeurfou** [comment]
It looks fair to me. The only really useful one are the one you mentioned in my opinion.

**Comment by jalvesaq** [comment]
Good! Thanks for your contribution. Now, I think it's time to close the issue.
